### ISSUE TITLE
macOS Support

### ISSUE THREAD

**Issue by Saadnajmi** [wcandillon/react-native-webgpu#92]
Hello, I'm the main maintainer for React Native macOS. I am interested in adding macOS support for this library, and wanted to start an issue to make sure this aligned with your vision for the repo. Basically two things would need to be done:

1) Add a test app with React Native Test App. This will help with maintaining the macOS test app because it simplifies down to a single pod file, I believe @okwasniewski was already thinking of doing the same for visionOS.

2) Update the sources for macOS. I looked at the repo and that diff doesn't look too hard. React Native macOS is largely compatible with React Native iOS code. A couple of places we need to call Appkit instead of UIKit mostly. I think Dawn should also compile for macOS so no issues there either :)

I am interested in submitting a PR, though it might be a while 😅

**Comment by wcandillon** [comment]
yes that sounds great. Adding the build for macOS shouldn't be a problem. And also adding an example app would be welcomed. I'm excited for this :)

**Comment by wcandillon** [comment]
Interestingly enough, there are already macOS binaries available at https://github.com/google/dawn/

**Comment by Saadnajmi** [comment]
@wcandillon Sorry, where are the binaries? I'm having trouble building dawn locally so I was seeing if I could just locally use some prebuilt binaries.

**Comment by wcandillon** [comment]
here they are: https://github.com/google/dawn/actions/workflows/ci.yml
Ours are at https://github.com/wcandillon/react-native-webgpu/actions/workflows/build-dawn.yml

I like the PR you opened, let's make it happen :)

**Comment by wcandillon** [comment]
the binaries are not available at https://github.com/wcandillon/react-native-webgpu/actions/runs/10744532243

**Comment by wcandillon** [comment]
Done :) We also added macOS support in our CI here and also for React Native Skia. We also plan to take things further by doing macOS e2e testing. Thank you for your help with this. We've been really happy with RNTA; our main hurdle there is the lack of Web support. Thank you again.
